ntbs-service-unit-tests/Services/NotificationServiceTest.cs

```diff
@@ -430,6 +431,36 @@ private void VerifyUpdateDatabaseCalled()
                 mock.SaveChangesAsync(It.IsAny<NotificationAuditType>(), It.IsAny<string>()));
         }
 
+        [Fact]
+        public async Task CreateNewNotificationForUserAsync_CreatesCaseUsingUserDetails()
+        {
+            // Arrange
+            const string serviceCode = "TBS0008";
+            const string username = "User1";
+            var userId = 1234;
+            var tbService = new TBService
+            {
+                ServiceAdGroup = "ServiceAshford",
+                Name = username,
+                Code = serviceCode
+            };
+            var mockPrincipal = new Mock<ClaimsPrincipal>();
+            mockPrincipal.Setup(u => u.FindFirst(ClaimTypes.Upn)).Returns(new Claim(ClaimTypes.Upn, username));
+            var user = new User { Id = userId, Username = username };
+
+            _mockUserService.Setup(c => c.GetDefaultTbService(mockPrincipal.Object)).Returns(Task.FromResult(tbService));
+            _mockReferenceDataRepository
+                .Setup(c => c.GetCaseManagersByTbServiceCodesAsync(new List<string> { serviceCode }))
+                .Returns(Task.FromResult((IList<User>)(new List<User> { user })));

```

Comment:
It might be better to include more than one user in the mock response, to validate that it's the logged in user that's being selected, rather than just the first or only one.

Reply:
Yep I agree, done!